corehq/apps/accounting/utils/unpaid_invoice.py

```diff
@@ -123,10 +123,19 @@ def _check_and_perform_action(cls, communication_model, context,
 
     @staticmethod
     def _should_send_invoice_reminder(communication_model, invoice):
-        return not communication_model.objects.filter(
-            invoice=invoice,
-            communication_type=CommunicationType.INVOICE_REMINDER,
-        ).exists()
+        if invoice.is_customer_invoice:
+            billing_record = invoice.customerbillingrecord_set.first()
```

Comment:
The way `BillingRecord` is written, it appears like its possible for an invoice to have multiple billing records. Do you know if this is the case? It's probably outside the scope of this PR, but if the intention was that they are 1-to-1 instead, we probably should have defined it as 1-to-1: https://docs.djangoproject.com/en/5.1/topics/db/queries/#one-to-one-relationships